hudi-hadoop-common/src/test/java/org/apache/hudi/common/bootstrap/index/TestBaseHoodieTableFileIndex.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.common.bootstrap.index;
+
+import org.apache.hudi.BaseHoodieTableFileIndex;
+import org.apache.hudi.common.config.HoodieCommonConfig;
+import org.apache.hudi.common.config.TypedProperties;
+import org.apache.hudi.common.engine.HoodieEngineContext;
+import org.apache.hudi.common.engine.HoodieLocalEngineContext;
+import org.apache.hudi.common.model.HoodieTableQueryType;
+import org.apache.hudi.common.table.HoodieTableMetaClient;
+import org.apache.hudi.common.testutils.HoodieCommonTestHarness;
+import org.apache.hudi.common.util.Option;
+import org.apache.hudi.storage.StoragePath;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+
+import static org.apache.hudi.hadoop.fs.HadoopFSUtils.getStorageConf;
+
+@ExtendWith(MockitoExtension.class)
+class TestBaseHoodieTableFileIndex extends HoodieCommonTestHarness {
+
+  @ParameterizedTest
+  @ValueSource(booleans = {false, true})
+  void refresh(boolean useSpillableMap) throws IOException {
+    initMetaClient();
+    TypedProperties properties = new TypedProperties();
+    if (useSpillableMap) {
+      properties.put(HoodieCommonConfig.HOODIE_FILE_INDEX_USE_SPILLABLE_MAP.key(), true);
+    }
+    BaseHoodieTableFileIndex baseHoodieTableFileIndex = new TestLocalIndex(
+        new HoodieLocalEngineContext(getStorageConf()),
+        metaClient,
+        properties,
+        HoodieTableQueryType.READ_OPTIMIZED,
+        Collections.emptyList(),
+        Option.empty(),
+        false,
+        true,
+        null,
+        true,
+        Option.empty(),
+        Option.empty()
+    );
+    Assertions.assertTrue(baseHoodieTableFileIndex.getFileSlicesCount() == 0);
```

Comment:
This does not seem to test any `refresh` functionality.

Reply:
Renamed the test. doRefresh is a protected method and no longer accessible from this unit test using Hadoop I/O libraries. 

doRefresh is still called from the constructor for BaseHoodieTableFileIndex and has coverage here.

Reply:
I see, got it.